My Home: Remove chevrons from internal links on My Home

p-jackson opened the pull request:
#### Changes proposed in this Pull Request

This is part of the short-term design changes being made to My Home pd2qGl-4i-p2

* Remove the chevron from the "Contact support" link in the "Get help" card
* Remove the chevron from the internal links in the "Quick links" card (the external links still show the external link indicator)

**Before**
<img width="797" alt="Screenshot 2021-06-10 at 2 45 51 PM" src="https://user-images.githubusercontent.com/1500769/121456770-af00bc00-c9fa-11eb-9615-f8e32c689a40.png">


**After**
<img width="797" alt="Screenshot 2021-06-10 at 2 46 18 PM" src="https://user-images.githubusercontent.com/1500769/121456757-a90adb00-c9fa-11eb-814a-5a73f66612a1.png">


#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Should be able to test on calypso.live
* Make sure no alignment has broken on mobile layouts

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #53141

p-jackson committed:
Remove chevron from "Contact support" link on My Home

p-jackson committed:
Remove chevron from interal quick links on My Home

Removing the `displayAsLink` prop from the <CompactCard> components
removed the chevron for most of the Quick Links. This prop was showing
the chevron for items that weren't real links (didn't have an `href`
prop).

For the internal links I added an opt-in prop (`hideLinkIndicator`) to
hide it. The reason I didn't blanket hide it for every item is because
we still want the link indicator appearing on external links.

autumnfjeld reviewed (changes requested):
✅  Desktop
![image](https://user-images.githubusercontent.com/5665959/121473534-2b4ccc80-ca06-11eb-95e4-9719518e0b93.png)


✅ Tablet
![image](https://user-images.githubusercontent.com/5665959/121473486-18d29300-ca06-11eb-9f9c-a63c64be6540.png)


🛑  Small screen:  Help fab button not aligned to the left on small screens
![image](https://user-images.githubusercontent.com/5665959/121473317-d5782480-ca05-11eb-9c8b-01afbe31195a.png)

Good to go after tiny tweek on small screens--note it was pre-existing, but if we can easily fix it would be coooolio.

p-jackson committed:
Vertically align left edge of all cards in mobile view

p-jackson commented:
I fixed up the alignment of the blue (?) icon on mobile. But then that put it out of alignment with the "go mobile" card below it, so I fixed that too. And then I noticed that the main task card at the top had different padding on mobile too ... 😅 

So it should all be consistent now: `24px` for regular layout, `16px` for phone.

https://user-images.githubusercontent.com/1500769/121477358-dd42c380-ca1b-11eb-91c1-bfd88c589ea7.mov

autumnfjeld reviewed (approved):
Thank you for fixing the little details!
